core-rust/core-api-server/src/core_api/conversions/substates/access_controller.rs

```diff
@@ -12,11 +13,95 @@ pub fn to_api_access_controller_substate(
     Ok(field_substate_versioned!(
         substate,
         AccessControllerFieldState,
-        substate => {
-            let data = scrypto_encode(substate).unwrap();
+        AccessControllerSubstate {
+            controlled_asset,
+            timed_recovery_delay_in_minutes,
+            recovery_badge,
+            state: (
+                primary_role_locking_state,
+                primary_role_recovery_attempt_state,
+                primary_role_badge_withdraw_attempt_state,
+                recovery_role_recovery_attempt_state,
+                recovery_role_badge_withdraw_attempt_state,
+            )
         },
         Value {
-            data_struct: Box::new(to_api_data_struct_from_bytes(context, &data)?),
+            controlled_vault_entity: Box::new(to_api_entity_reference(
```

Comment:
This should be an owned entity, rather than a reference

Reply:
We don't actually have a way (in the Core API) to indicate this difference.
There was some idea to do it, but a PR was abandoned, since it did not solve any client problem - see https://github.com/radixdlt/babylon-node/pull/600#issuecomment-1687816432

But let me apply the weak naming convention that we have for "owned vaults" (see `rewards_vault`, `stake_xrd_vault` and more): drop the `_entity` 👍